### PR TITLE
fix: Assignment reminder modal appears when deleting a team event type

### PR DESCRIPTION
**Pull request by anikdhabal**
## What does this PR do?

- Fixes #XXXX (GitHub issue number)
- Fixes CAL-XXXX (Linear issue number - should be visible at the bottom of the GitHub issue description)

Problem:-
Screenshot 2024-06-23 171204

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected)
- [ ] I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com)
- [ ] I have added or modified automated tests that prove my fix is effective or that my feature works (PRs might be rejected if logical changes are not properly tested)

## How should this be tested?

- Are there environment variables that should be set?
- What are the minimal test data to have?
- What is expected (happy path) to have (input and output)?
- Any other important info that could help to test that PR

## Checklist

- I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my changes generate no new warnings

### CODE REVIEW

File: apps/web/components/eventtype/EventTypeSingleLayout.tsx [hunk 1]
```diff
@@ -125,7 +125,7 @@ function DeleteDialog({
     onSuccess: async () => {
       await utils.viewer.eventTypes.invalidate();
       showToast(t("event_type_deleted_successfully"), "success");
-      router.push("/event-types");
+      router.push("/event-types?");
```
Comment (hariombalhara) [inline apps/web/components/eventtype/EventTypeSingleLayout.tsx:128]: Why this change?

File: apps/web/modules/event-types/views/event-types-single-view.tsx [hunk 2]
```diff
@@ -415,7 +415,7 @@ const EventTypePage = (props: EventTypeSetupProps) => {
         !leaveWithoutAssigningHosts.current &&
         !!team &&
         (hosts.length === 0 || assignedUsers.length === 0) &&
-        (url === "/event-types" || paths[1] !== "event-types")
+        (url === "/event-types" || (url !== "/event-types?" && paths[1] !== "event-types"))
```
Comment (hariombalhara) [inline apps/web/modules/event-types/views/event-types-single-view.tsx:418]: We are trying to differentiate b/w "navigating to `/event-types` when the event-type is deleted" and "a regular navigation to the page"

Instead of indirectly tackling this by adding `?` we can directly use the delete status

**Review by hariombalhara — CHANGES_REQUESTED** [review]
Thanks for the PR @anikdhabal 🙏

I have suggested a different approach to solve the problem, let me know how that sounds.

**Comment by anikdhabal** [comment]
@hariombalhara thanks for the review. I've updated the code.

**Comment by anikdhabal** [comment]
@hariombalhara check this one

File: apps/web/components/eventtype/EventTypeSingleLayout.tsx [hunk 3]
```diff
@@ -117,14 +119,19 @@ function DeleteDialog({
   eventTypeId,
   open,
   onOpenChange,
-}: { isManagedEvent: string; eventTypeId: number } & Pick<DialogProps, "open" | "onOpenChange">) {
+  isEventTypeDeleted,
```
Comment (hariombalhara) [inline apps/web/components/eventtype/EventTypeSingleLayout.tsx:122]: `isEventTypeDeleted` is a reference that is being updated in this function. This isn't a good coding practice. I would recommend adding a `onDelete` prop instead.

File: apps/web/components/eventtype/EventTypeSingleLayout.tsx [hunk 4]
```diff
@@ -181,6 +188,7 @@ function EventTypeSingleLayout({
   isUserOrganizationAdmin,
   bookerUrl,
   activeWebhooksNumber,
+  isEventTypeDeleted,
```
Comment (hariombalhara) [inline apps/web/components/eventtype/EventTypeSingleLayout.tsx:191]: Similarly here.

File: apps/web/modules/event-types/views/event-types-single-view.tsx [hunk 5]
```diff
@@ -416,7 +417,8 @@ const EventTypePage = (props: EventTypeSetupProps) => {
         !leaveWithoutAssigningHosts.current &&
         !!team &&
         (hosts.length === 0 || assignedUsers.length === 0) &&
-        (url === "/event-types" || paths[1] !== "event-types")
+        (url === "/event-types" || paths[1] !== "event-types") &&
+        !isEventTypeDeleted.current
```
Comment (hariombalhara) [inline apps/web/modules/event-types/views/event-types-single-view.tsx:421]: For code brevity, better to early return if eventType is deleted.

**Review by hariombalhara — CHANGES_REQUESTED** [review]
Suggested some more improvements

**Review by hariombalhara — APPROVED** [review]
LGTM now !! Made a small change.
